R/polars_envvars.R

```diff
@@ -0,0 +1,138 @@
+#' * `POLARS_STREAMING_CHUNK_SIZE` (`<variable>`): Chunk size used in the
+#'   streaming engine. Integer larger than 1. By default, the chunk size is
+#'   determined by the schema and size of the thread pool. For some datasets
+#'   (esp. when you have large string elements) this can be too optimistic and
+#'   lead to Out of Memory errors.
+#' * `POLARS_TABLE_WIDTH` (`<variable>`): Set the maximum width of a table in
+#'   characters.
+#' * `POLARS_VERBOSE` (`"0"`): Enable additional verbose/debug logging.
+#' * `POLARS_WARN_UNSTABLE` (`"0"`): Issue a warning when unstable functionality
+#'    is used. Enabling this setting may help avoid functionality that is still
+#'    evolving, potentially reducing maintenance burden from API changes and bugs.
+#'    Can be `"0"` or `"1"`.
+#'
+# See https://github.com/pola-rs/polars/blob/6913b16c255f38c8609ea11f1b2edab6482cedc5/py-polars/polars/config.py#L73 # nolint
+#' The following configuration options are present in the Python API but
+#' currently cannot be changed in R: decimal separator, thousands separator,
+#' float precision, float formatting, trimming decimal zeros.
+#'
+#' @return `polars_envvars()` returns a named list where the names are the names
+#'   of environment variables and values are their values.
+#'
+#' @export
+#' @examples
+#' polars_envvars()
+#'
+#' # By default, long string values are truncated.
+#' df <- pl$DataFrame(x = "This is a very very very long sentence.")
+#' df
+#'
+#' # We can temporarily allow for wider columns with `withr::with_envvar()`:
+#' withr::with_envvar(
+#'   list(POLARS_FMT_STR_LEN = "50"),
+#'   print(df)
+#' )
+#'
+#' # Or we could set it permanently with `Sys.setenv(POLARS_FMT_STR_LEN = "50")`.
+polars_envvars <- function() {
+  envvars <- rbind(
+    c("POLARS_FMT_MAX_COLS", "5"),
+    c("POLARS_FMT_MAX_ROWS", "8"),
+    # Exist in polars but can't be set (even in py-polars)
+    # c("POLARS_FMT_NUM_DECIMAL", ""),
+    # c("POLARS_FMT_NUM_GROUP_SEPARATOR", ""),
+    # c("POLARS_FMT_NUM_LEN", ""),
+    c("POLARS_FMT_STR_LEN", "32"),
+    c("POLARS_FMT_TABLE_CELL_ALIGNMENT", "LEFT"),
+    c("POLARS_FMT_TABLE_CELL_LIST_LEN", "3"),
+    c("POLARS_FMT_TABLE_CELL_NUMERIC_ALIGNMENT", "LEFT"),
+    c("POLARS_FMT_TABLE_DATAFRAME_SHAPE_BELOW", "0"),
+    c("POLARS_FMT_TABLE_FORMATTING", "UTF8_FULL_CONDENSED"),
+    c("POLARS_FMT_TABLE_HIDE_COLUMN_DATA_TYPES", "0"),
+    c("POLARS_FMT_TABLE_HIDE_COLUMN_NAMES", "0"),
+    c("POLARS_FMT_TABLE_HIDE_COLUMN_SEPARATOR", "0"),
+    c("POLARS_FMT_TABLE_HIDE_DATAFRAME_SHAPE_INFORMATION", "0"),
+    c("POLARS_FMT_TABLE_INLINE_COLUMN_DATA_TYPE", "0"),
+    c("POLARS_FMT_TABLE_ROUNDED_CORNERS", "0"),
+    c("POLARS_MAX_THREADS", thread_pool_size()),
+    c("POLARS_STREAMING_CHUNK_SIZE", "variable"),
+    c("POLARS_TABLE_WIDTH", "variable"),
+    c("POLARS_VERBOSE", "0"),
+    c("POLARS_WARN_UNSTABLE", "0")
+  ) |>
+    as.data.frame()
+
+  out <- vector("list", length(envvars))
```

Comment:
This seems like a pretty roundabout way to do this when building a list, could you please fix that?

Reply:
Done in [866c898](https://github.com/pola-rs/r-polars/pull/1580/commits/866c898d4ce986c65d03a85dc4be2291b5d58173)